src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1185,8 +1186,58 @@ export default class SourceFileLinter {
 		].includes(nodeType);
 	}
 
+	analyzeExportedValuesByLib(node: ts.PropertyAccessExpression) {
+		if (node.name?.kind !== ts.SyntaxKind.Identifier) {
+			return;
+		}
+
+		let exprNode = node.expression;
+		const namespace: string[] = [];
+		while (ts.isPropertyAccessExpression(exprNode)) {
+			namespace.unshift(exprNode.name.text);
+			exprNode = exprNode.expression;
+		}
+		const exprType = this.checker.getTypeAtLocation(exprNode);
+		const potentialLibImport = exprType.symbol?.getName().replaceAll("\"", "") ?? "";
+
+		// Checks if the left hand side is a library import.
+		// It's sufficient just to check for "/library" as the end of the string by convention
+		if (!potentialLibImport.endsWith("/library")) {
+			return;
+		}
+
+		const moduleName = [
+			potentialLibImport.replace("/library", ""),
+			...namespace,
+			node.name.text,
+		].join("/");
+
+		// Check if the module is registered within ambient modules
+		const ambientModules = this.checker.getAmbientModules();
+		const isRegisteredAsUi5Module = ambientModules.some((module) =>
+			module.name === `"${moduleName}"`);
+
+		// Check if it has been imported as UI5 module
```

Comment:
It is irrelevant whether the module has also been imported. In UI5 v1.x this ensures that the value via the implicit global access is available, but in v2.x the global won't be available anymore at all, as no globals are exported by the UI5 framework (e.g. via extend call or sap.ui.define bExport).

Reply:
Thanks!